### PR TITLE
Introduce Changelog in build pipeline

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added CHANGELOG.md document to the repository.
- Added CHANGELOG_TEMPLATE.md document to the repository.

### Changed
- Updated the Pull Request description section in CONTRIBUTING.md.
- Extended the build to update the content of the CHANGELOG.md with the Pull Request description.

### Removed
- Removed the 'dev' branch from .travis.yml.

### CODE REVIEW

File: Build/build.ps1 [hunk 1]
```diff
@@ -15,22 +15,158 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
...
+    [CmdletBinding()]
+    [OutputType([string])]
+
+    $searchIssuesAPIUrl = 'https://api.github.com/search/issues'
+    $repository = 'vmware/dscr-for-vmware'
+    $searchType = 'pr'
+    $pullRequestState = 'closed'
+
+    $base64Token = [System.Convert]::ToBase64String([char[]] $env:GH_TOKEN)
+    $uri = "$searchIssuesAPIUrl?q=repo:$repository+type:$searchType+state:$pullRequestState+$env:TRAVIS_COMMIT is:merged"
```
Comment (dmilov) [inline Build/build.ps1]: There is no max items limit? Is it going to return all closed pull requests?

File: Build/build.ps1 [hunk 2]
```diff
@@ -15,22 +15,158 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
...
+        [string] $PullRequestDescription,
+
+        [Parameter(Mandatory = $true)]
+        [string] $ModuleVersion
+    )
+
+    $changelogDocument = Get-Content -Path $ChangelogDocumentPath
+
+    # The Header in the CHANGELOG.md consists of the first two lines plus one empty line so we need the first three lines of the file.
+    $headerLength = 2
```
Comment (dmilov) [inline Build/build.ps1]: Hm, hardcoded number of lines for the header is tricky. More relieble will be to search for the top changelog record and insert the new records on top of it.

File: Build/build.ps1 (hunk 1 above)
Comment (SimeonGerginov) [reply Build/build.ps1]: Well in the uri "$env:TRAVIS_COMMIT" is used so it will return only one Pull Request.

File: Build/build.ps1 (hunk 2 above)
Comment (SimeonGerginov) [reply Build/build.ps1]: Done.

**Pull request merged**